### PR TITLE
Removing box from arm without hand and bimanual system without hands

### PR DESCRIPTION
**Pull request by jeferal**
## Proposed changes

This pull request aims to remove the box from the simulation of the arm without the hand. The file sr_ur_arm_box.launch has been replaced by sr_ur_arm.launch and config files for a single ur have been created. They are used by sr_simulation_control_loop.launch and sr_hardware_control_loop.launch. When launching sr_ur_arm.launch, the xacro sr_ur.urdf.xacro is called and this one is parametrized. This has been tested under simulation and with the real robot.

Launch single robot without hand:
```bash
roslaunch sr_robot_launch sr_ur_arm.launch sim:=true
```

Another objective of this PR is to remove the hands from the bimanual system. Before setting the parameter hands to 'none' spawned both hands on the simulation because the file sr_ur_arms_hands.launch called sr_bimanual_simulation_control_loop.launch setting hands to both by default. This works now under simulation and has been tested with the two real ur's.
Finally, it is also possible to spawn a single hand in the bimanual system (right or left) under simulation.

bimanual

It does not work on the real robots due to a missmatch between the parameter names when autodetection is used (It did not work before either).

Launch both ur's with right hand:
```bash
roslaunch sr_robot_launch sr_bimanual_ur10amrs_hands.launch sim:=true hands:=right
```

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [x] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [x] Manually tested on hardware (if hardware specific or related).

## Documentation

- [ ] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

**Review by bdenoun — APPROVED** [review]

**Review by maxzieba — COMMENTED** [review]
Just a kind request to change the title to have a nice changelog :)

File: sr_multi_description/urdf/bimanual_srhand_ur.urdf.xacro [hunk 1]
```diff
@@ -61,23 +62,31 @@
...
+                    fingers="$(arg right_fingers)" tip_sensors="$(arg right_tip_sensors)" mid_sensors="$(arg right_mid_sensors)"
+                    prox_sensors="$(arg right_prox_sensors)" palm_sensor="$(arg right_palm_sensor)"/>
+    </xacro:if>
 
-    <joint name="ra_arm_to_hand" type="fixed">
-        <parent link="ra_flange"/>
-        <child link="rh_forearm"/>
-        <origin xyz="0.012 0.0 0.0" rpy="${pi/2.0} ${pi} ${pi/2.0}"/>
-    </joint>
+    <xacro:if value="${'$(arg hands)' == 'both' or '$(arg hands)' == 'right'}">
```
Comment (beatrizleon) [inline sr_multi_description/urdf/bimanual_srhand_ur.urdf.xacro]: As both if are the same here and in the section below I would group them into the same if statement

File: sr_multi_moveit/sr_multi_moveit_config/config/robot_configs/bimanual_shl_ur10e.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+robot:
```
Comment (beatrizleon) [inline sr_multi_moveit/sr_multi_moveit_config/config/robot_configs/bimanual_shl_ur10e.yaml]: Should we name this file: bimanual_sh_left_ur10e.yaml  and the other bimanual_sh_right_ur10e.yaml? I think it looks cleaner

File: sr_multi_moveit/sr_multi_moveit_config/config/robot_configs/left_ur10.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        moveit_path:
+          package: sr_multi_moveit_config
+          relative_path: /config/ur
+        extra_groups_config_path: /config
+        group_states:
+          - home
+          - up
+        hand: False
+
+ra_sr_ur_robot_hw:
```
Comment (beatrizleon) [inline sr_multi_moveit/sr_multi_moveit_config/config/robot_configs/left_ur10.yaml]: should be la
```suggestion
la_sr_ur_robot_hw:
```

File: sr_multi_moveit/sr_multi_moveit_config/config/robot_configs/left_ur10e.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        moveit_path:
+          package: sr_multi_moveit_config
+          relative_path: /config/ur
+        extra_groups_config_path: /config
+        group_states:
+          - home
+          - up
+        hand: False
+
+ra_sr_ur_robot_hw:
```
Comment (beatrizleon) [inline sr_multi_moveit/sr_multi_moveit_config/config/robot_configs/left_ur10e.yaml]:
```suggestion
la_sr_ur_robot_hw:
```

File: sr_multi_moveit/sr_multi_moveit_config/config/robot_configs/left_ur5.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        moveit_path:
+          package: sr_multi_moveit_config
+          relative_path: /config/ur
+        extra_groups_config_path: /config
+        group_states:
+          - home
+          - up
+        hand: False
+
+ra_sr_ur_robot_hw:
```
Comment (beatrizleon) [inline sr_multi_moveit/sr_multi_moveit_config/config/robot_configs/left_ur5.yaml]:
```suggestion
la_sr_ur_robot_hw:
```

File: sr_multi_moveit/sr_multi_moveit_config/config/robot_configs/left_ur5e.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        moveit_path:
+          package: sr_multi_moveit_config
+          relative_path: /config/ur
+        extra_groups_config_path: /config
+        group_states:
+          - home
+          - up
+        hand: False
+
+ra_sr_ur_robot_hw:
```
Comment (beatrizleon) [inline sr_multi_moveit/sr_multi_moveit_config/config/robot_configs/left_ur5e.yaml]:
```suggestion
la_sr_ur_robot_hw:
```

File: sr_robot_launch/launch/load_robot_description_bimanual.launch [hunk 7]
```diff
@@ -57,10 +57,9 @@
...
-  <arg if="$(eval arms and not arg('hands') == 'none')" name="robot_description" default="'$(find sr_multi_description)/urdf/bimanual_srhand_ur.urdf.xacro'"/>
-
+  <arg if="$(eval arms and not arg('hands') == 'none')" name="robot_description" default="'$(find sr_multi_description)/urdf/bimanual_srhand_ur.urdf.xacro'"/>  
   <arg if="$(eval arms and arg('hands') == 'none')" name="load_robot_description_command" value="xacro $(arg robot_description) robot_model:=$(arg robot_model) arm_1_z:=$(arg arm_1_z) arm_2_z:=$(arg arm_2_z) arm_x_separation:=$(arg arm_x_separation) arm_y_separation:=$(arg arm_y_separation)"/>
-  <arg if="$(eval arms and not arg('hands') == 'none')" name="load_robot_description_command" value="xacro $(arg robot_description) robot_model:=$(arg robot_model) right_hand_type:=$(arg right_hand_type) right_hand_version:=$(arg right_hand_version)
+  <arg if="$(eval arms and not arg('hands') == 'none')" name="load_robot_description_command" value="xacro $(arg robot_description) hands:=$(arg hands) robot_model:=$(arg robot_model) right_hand_type:=$(arg right_hand_type) right_hand_version:=$(arg right_hand_version)
```
Comment (beatrizleon) [inline sr_robot_launch/launch/load_robot_description_bimanual.launch:62]: I don't understand why only the right hand is passed here?

File: sr_robot_launch/launch/sr_bimanual_hardware_control_loop.launch [hunk 8]
```diff
@@ -68,7 +68,8 @@
   <arg if="$(eval not hands == 'none' and arg('left_hand_type') == 'hand_e')" name="left_arm_payload_mass" default="5.0"/>
   <arg if="$(eval not hands == 'none' and arg('left_hand_type') == 'hand_g')" name="left_arm_payload_mass" default="2.7"/>
   <arg unless="$(eval hands == 'none')" name="arm_payload_cog" default="[0.0, 0.0, 0.12]"/>
-  <arg if="$(eval hands == 'none')" name="arm_payload_mass" default="0.0"/>
+  <arg if="$(eval hands == 'none')" name="right_arm_payload_mass" default="0.0"/>
```
Comment (beatrizleon) [inline sr_robot_launch/launch/sr_bimanual_hardware_control_loop.launch]: what if hands=='right' ? shouldn't we also be setting this?

File: sr_robot_launch/launch/sr_bimanual_simulation_control_loop.launch [hunk 9]
```diff
@@ -66,11 +66,15 @@
...
   <!-- ROBOT CONFIG FILES-->
   <!-- Hands and arms -->
-  <arg name="robot_config_file" if="$(eval not arg('hands') == 'none' and arg('arm_ctrl'))" default="$(find sr_multi_moveit_config)/config/robot_configs/bimanual_sh_$(arg robot_model).yaml"/>
+  <arg name="robot_config_file" if="$(eval arg('hands') == 'both' and arg('arm_ctrl'))" default="$(find sr_multi_moveit_config)/config/robot_configs/bimanual_sh_$(arg robot_model).yaml"/>
   <!-- Arms but no hands -->
   <arg name="robot_config_file" if="$(eval arg('hands') == 'none' and arg('arm_ctrl'))" default="$(find sr_multi_moveit_config)/config/robot_configs/bimanual_$(arg robot_model).yaml"/>
   <!-- Hands but no arms -->
   <arg name="robot_config_file" if="$(eval not arg('hands') == 'none' and not arg('arm_ctrl'))" default="$(find sr_multi_moveit_config)/config/robot_configs/bimanual_sh.yaml"/>
+  <!-- Arms and right hand only -->
+  <arg name="robot_config_file" if="$(eval arg('hands') == 'right' and arg('arm_ctrl'))" default="$(find sr_multi_moveit_config)/config/robot_configs/bimanual_shr_$(arg robot_model).yaml"/>
```
Comment (beatrizleon) [inline sr_robot_launch/launch/sr_bimanual_simulation_control_loop.launch]: if the name of the file is changed, we need to fix it here

**Review by beatrizleon — CHANGES_REQUESTED** [review]

File: sr_robot_launch/launch/load_robot_description_bimanual.launch (hunk 7 above)
Comment (jeferal) [reply sr_robot_launch/launch/load_robot_description_bimanual.launch:62]: I think github hid these lines. They should appear, the left hand is passed from 67 to 70

Comment (beatrizleon) [reply sr_robot_launch/launch/load_robot_description_bimanual.launch:62]: should we be passing here depending on the side? (i.e. if both then both, if right only right and left only left)?

File: sr_robot_launch/launch/sr_bimanual_hardware_control_loop.launch [hunk 10]
```diff
@@ -63,27 +63,35 @@
   <arg name="left_arm_speed_scale" default="0.5"/>
 
   <!-- Sets up right payload -->
```
Comment (beatrizleon) [inline sr_robot_launch/launch/sr_bimanual_hardware_control_loop.launch]:
```suggestion
  <!-- Sets up right payload and cog-->
```

File: sr_robot_launch/launch/sr_bimanual_hardware_control_loop.launch [hunk 11]
```diff
@@ -63,27 +63,35 @@
...
-  <arg unless="$(eval hands == 'none')" name="arm_payload_cog" default="[0.0, 0.0, 0.12]"/>
-  <arg if="$(eval hands == 'none')" name="arm_payload_mass" default="0.0"/>
-  <arg if="$(eval hands == 'none')" name="arm_payload_cog" default="[0.0, 0.0, 0.0]"/>
+  <arg if="$(eval (hands == 'both' or hands == 'right') and arg('right_hand_type') == 'hand_e')" name="right_arm_payload_mass" default="5.0"/>
+  <arg if="$(eval (hands == 'both' or hands == 'right') and arg('right_hand_type') == 'hand_g')" name="right_arm_payload_mass" default="2.7"/>
+  <arg if="$(eval (hands == 'both' or hands == 'left') and arg('left_hand_type') == 'hand_e')" name="left_arm_payload_mass" default="5.0"/>
+  <arg if="$(eval (hands == 'both' or hands == 'left') and arg('left_hand_type') == 'hand_g')" name="left_arm_payload_mass" default="2.7"/>
+  <arg if="$(eval hands == 'none' or hands == 'left')" name="right_arm_payload_mass" default="0.0"/>
+  <arg if="$(eval hands == 'none' or hands == 'right')" name="left_arm_payload_mass" default="0.0"/>
+
```
Comment (beatrizleon) [inline sr_robot_launch/launch/sr_bimanual_hardware_control_loop.launch]: no need for this empty line

File: sr_robot_launch/launch/sr_ur_arms_hands.launch [hunk 12]
```diff
@@ -59,11 +59,20 @@
...
+  <arg if="$(eval hand_ctrl and arm_trajectory and hands=='right')" name="trajectory_controllers_list" 
+       default="[/ra_trajectory_controller/state, /la_trajectory_controller/state, /rh_trajectory_controller/state]"/>
+
+  <arg if="$(eval hand_ctrl and arm_trajectory and hands=='left')" name="trajectory_controllers_list" 
+       default="[/ra_trajectory_controller/state, /la_trajectory_controller/state, /lh_trajectory_controller/state]"/>
+  
   <arg if="$(eval not hand_ctrl and not arm_trajectory)" name="trajectory_controllers_list" value="[]"/>
-  <arg if="$(eval hand_ctrl and not arm_trajectory)" name="trajectory_controllers_list"
+  
+  <arg if="$(eval hand_ctrl and not arm_trajectory and hands=='both')" name="trajectory_controllers_list"
```
Comment (beatrizleon) [inline sr_robot_launch/launch/sr_ur_arms_hands.launch:73]: I think this is the none case
```suggestion
  <arg if="$(eval hand_ctrl and not arm_trajectory and hands=='none')" name="trajectory_controllers_list"
```

File: sr_robot_launch/launch/load_robot_description_bimanual.launch (hunk 7 above)
Comment (jeferal) [reply sr_robot_launch/launch/load_robot_description_bimanual.launch:62]: I also pass to the xacro now the hands argument and so if hands=right, although the left arguments are passed, they will not be used inside the xacro. So I think that condition works for both, right and left cases (if I'm not wrong). In my opinion adding the if both... would make the code longer, but less confusing indeed.

Comment (beatrizleon) [reply sr_robot_launch/launch/load_robot_description_bimanual.launch:62]: yes, this sounds good

**Review by beatrizleon — APPROVED** [review]

**Review by giusebar — APPROVED** [review]

**Pull request merged**